internal/ghworkflow/workflow_auto-update.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 SAP SE
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// limitations under the License.
+
+package ghworkflow
+
+import "github.com/sapcc/go-makefile-maker/internal/core"
+
+func autoUpdateWorkflow(cfg *core.GithubWorkflowConfiguration) error {
+	w := newWorkflow("go-makefile-maker", cfg.Global.DefaultBranch, nil)
+	w.Permissions.Contents = tokenScopeWrite // for peter-evans/create-pull-request to create commit & PR
+
+	w.On.WorkflowDispatch = workflowDispatch{manual: "true"}
+	w.On.PullRequest = pushAndPRTriggerOpts{}
+	w.On.Schedule = []cronExpr{{Cron: "00 07 * * 1"}} // every Monday at 07:00 AM
+
+	j := baseJob("go-makefile-maker")
+	j.addStep(jobStep{
+		Name: "Install go-makefile-maker",
+		Run:  "go install github.com/sapcc/go-makefile-maker@latest",
+	})
+	j.addStep(jobStep{
+		Name: "Run go-makefile-maker",
+		Run:  "go-makefile-maker",
+	})
+
+	j.addStep(jobStep{
+		Name: "Create Pull Request",
+		Uses: "peter-evans/create-pull-request@v4",
+		With: map[string]interface{}{
+			"assignees": "SuperSandro2000",
```

Comment:
We should not hardcode names here. While this tool is primarily used inside of sapcc, if I remember correctly, I also used it for some private stuff already.